specs/PushNotifications/ToastNotifications-spec.md

```diff
@@ -0,0 +1,551 @@
+        High, // The notification should be treated as high priority. For desktop PCs, this means during connected standby mode the incoming notification can turn on the screen for Surface-like devices if it doesn't have a closed lid detected.
+    };
+
+    runtimeclass AppNotification
+    {
+        AppNotification(Windows.Data.Xml.Dom.XmlDocument payload);
+
+        // Unique identifier used to replace a notification within a group.
+        String Tag;
+
+        // Unique identifier for a Notification group in the app
+        String Group;
+
+        // A unique identifier for the Notification generated by the platform.
+        UInt32 Id { get; };
+
+        // The notification Xml Payload
+        Windows.Data.Xml.Dom.XmlDocument Payload{ get; };
+
+        // Gets or sets additional information about the Notification progress.
+        AppNotificationProgressData Progress;
+
+        // Gets or sets the time after which a Notification should not be displayed.
+        Windows.Foundation.DateTime Expiration;
+
+        // Indicates whether the Notification will remain in the Action Center after a reboot.
+        Boolean ExpiresOnReboot;
+
+        // Gets or sets the priority for a Notification.
+        // Hints on how and at what urgency level a notification should be presented to the user (whether to wake up the screen, etc).
+        AppNotificationPriority Priority;
+
+        // Gets or sets whether a Notification's pop-up UI is displayed on the user's screen.
+        Boolean SuppressDisplay;
+    }
+
+    // The manager class which encompasses all App Notification API Functionality
+    [experimental]
+    runtimeclass AppNotificationManager
+    {
+        // Gets a Default instance of a AppNotificationManager
+        static AppNotificationManager Default{ get; };
+
+        // Register an activator using an ActivationInfo context and caches the token for unregister
+        void RegisterActivator(AppNotificationActivationInfo details);
+
+        // Unregisters the activator and removes the cached Registration token.
+        void UnregisterActivator();
+
+        // Event handler for Notification Activations
+        event Windows.Foundation.TypedEventHandler<Microsoft.Windows.AppLifecycle.ExtendedActivationKind, AppNotificationActivatedEventArgs> AppNotificationActivated;
+
+        // Displays the Notification in Action Center
+        void Show(AppNotification notification);
+
+        // Updates the Notification for a Progress related operation using Tag and Group
+        Windows.Foundation.IAsyncOperation<AppNotificationProgressResult> UpdateProgressDataAsync(AppNotificationProgressData data, String tag, String group);
+
+        // Updates the Notification for a Progress related operation using Tag
+        Windows.Foundation.IAsyncOperation<AppNotificationProgressResult> UpdateProgressDataAsync(AppNotificationProgressData data, String tag);
```

Comment:
We can remove notifications by Tag, Group, or Tag+Group.

We can update a notification by Tag, Tag+Group, but not Group alone.

Why the mismatch?

Reply:
Because you can have multiple notifications in the Action Centre today with the same Group. A toast that attempts to replace every item in that group is an unsupported scenario. So the developer is expected to provide: tag, or tag + group for operations like replace or update to work.